ServerTesterClient/src/Client.java

```diff
@@ -14,97 +14,241 @@
 	private static final String MESSAGE_TYPE_LOGIN = "LOGIN";
 
 	private static final String MESSAGE_TYPE_TEXT = "TEXT";
+
+	public ClientRegisterLoginAndMessageTest test;
 	
 	public static void main(String args[]) throws URISyntaxException {
-		Client c = new Client(new URI("ws://localhost:1234"));
-		c.connect();
+
+		System.out.println("Scenario 1 - client 1 and 2 register, login, then client 1 sends client 2 a message while both are logged in.\n");
+		Client1 c1 = new Client1(new URI("ws://localhost:1234"));
+		System.out.println("Client 1 connecting");
+		c1.connect();
+		
+		try {
+			Thread.sleep(500);
```

Comment:
You can use ExecutorService to wait exactly for the amount of time it needs (but less than a specific timeout). You can find the code here: #21 -- Lines 66 - 86

Reply:
good point. I also want to test the order of specific things happening. Should I have an executor service for every single one of those situations?

Reply:
I think you can use the same executor instance but make sure you submit a task _after_ the previous one has executed (i.e. after try-catch with timeout). When you submit a task, you can override the call (each time) to do a specific action and can attach it to the same Future instance.

Reply:
As these are just used for tests, it's okay like this for now